packages/components/src/snackbar/types.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import type {
+	KeyboardEvent,
+	MouseEvent,
+	MutableRefObject,
+	ReactNode,
+} from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import type { WPElement } from '@wordpress/element';
+
+export type ButtonEvent =
+	| KeyboardEvent< HTMLButtonElement >
+	| MouseEvent< HTMLButtonElement >;
+
+export type Action = {
+	label: string;
+	url: string;
+	onClick?: ( event: ButtonEvent ) => void;
+};
+
+export type Notice = {
```

Comment:
This is really [WPNotice](https://github.com/WordPress/gutenberg/blob/62987d7e995a6d24f6b7d5a491c73aff5a1097c8/packages/notices/src/store/selectors.js#L20).

But I couldn't find a good way to import from `packages/notices/`, given that it's JSDoc, and it doesn't export a type.

Reply:
Good idea, I committed the TODO in https://github.com/WordPress/gutenberg/pull/45472/commits/5636934be887f7efdc19536ec4e6f5c749ab15fa

Reply:
Apologies for not being 100% clear — I was proposing to add the TODO comment in `packages/components/src/snackbar/types.ts`, above the `NoticeAction` type. We don't want to touch the types in `@wordpress/notices`, because those types are for the data store, and should therefore not be related to the component.


Would you mind moving the comment over? Thank you!

Reply:
Ah, sorry. Moved it in https://github.com/WordPress/gutenberg/pull/45472/commits/9af48f2ed27ce98a5e531f05909a85dcc7aab75b